crates/rpc/rpc-api/src/optimism.rs

```diff
@@ -0,0 +1,204 @@
+#![allow(missing_docs)]
+#![allow(unreachable_pub)]
+use jsonrpsee::{core::RpcResult, proc_macros::rpc};
+use reth_primitives::{Address, BlockId, BlockNumber, ChainId, Genesis, B256, U256, U64};
+use reth_rpc_types::BlockNumberOrTag;
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, net::IpAddr, time::Duration};
+
+/// todo: move to reth_rpc_types
+
+/// https://github.com/ethereum-optimism/optimism/blob/develop/op-service/eth/id.go#L33
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct L2BlockRef {
+    pub hash: B256,
+    pub number: BlockNumber,
+    pub parent_hash: B256,
+    pub timestamp: U64,
+    pub l1origin: BlockId,
+    pub sequence_number: u64,
+}
+
+/// https://github.com/ethereum-optimism/optimism/blob/develop/op-service/eth/id.go#L52
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct L1BlockRef {
+    pub hash: B256,
+    pub number: BlockNumber,
+    pub parent_hash: B256,
+    pub timestamp: U64,
```

Comment:
should also use primitive u64 with alloy_serde::u64_hex

Reply:
Having turn it to u64,  but `alloy_serde::u64_hex` should not be needed. 

https://github.com/AbnerZheng/reth/blob/846977e911e834452827bace98ad31bb5903d98b/crates/rpc/rpc-api/src/optimism.rs#L332-L337